### PR TITLE
[Merged by Bors] - feat: `NormedAlgebra.complexToReal` and other `RestrictScalars` instances

### PR DESCRIPTION
**Pull request by j-loreaux**
This adds some instances regarding normed algebra structures to `RestrictScalars` to mimic those for normed spaces.

In addition, given a normed `ℂ`-algebra, this puts a normed `ℝ`-algebra structure on the same space. This is *not* adding any new data instances, as these already exist from `Algebra.complexToReal`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Basic.lean [hunk 1]
```diff
@@ -425,6 +422,43 @@ instance {𝕜 : Type*} {𝕜' : Type*} {E : Type*} [I : NormedAddCommGroup E] :
...
+    NonUnitalNormedCommRing (RestrictScalars 𝕜 𝕜' E) :=
+  I
+
+instance {𝕜 : Type*} {𝕜' : Type*} {E : Type*} [I : SeminormedCommRing E] :
+    SeminormedCommRing (RestrictScalars 𝕜 𝕜' E) :=
+  I
+
+instance {𝕜 : Type*} {𝕜' : Type*} {E : Type*} [I : NormedCommRing E] :
+    NormedCommRing (RestrictScalars 𝕜 𝕜' E) :=
+  I
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/Basic.lean:455]: Please use `variable` to deduplicate the parameters.

**Comment by jcommelin** [comment]
bors d+

**Comment by j-loreaux** [comment]
bors merge

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**